ext/odbc/php_odbc.c

```diff
@@ -947,7 +987,7 @@ PHP_FUNCTION(odbc_prepare)
 		}
 	}
 
-	zend_hash_next_index_insert_new(&ODBCG(results), return_value);
+	odbc_insert_new_result(return_value);
```

Comment:
I realized that `zend_hash_next_index_insert_new()` alone won't increase the refcount of the objects, which would lead to failures. That's why I extracted this function call along with a `Z_TRY_ADDREF_P()` to a function because this repeated many times